src/Search.hs

```diff
@@ -0,0 +1,57 @@
+module Search
+    ( searchBoards
+    ) where
+
+import Control.Monad (mzero)
+import Data.Maybe (fromMaybe)
+import Data.List as List
```

Comment:
It's interesting that you need to qualify the import of `Data.List` -- where else does `filter` come from? I'd qualify the _other_ filter, since `Data.List` is usually assumed to be the one you're using when you use `filter`.